### PR TITLE
ensure that the bias is also in the correct dtype

### LINKED ISSUE #1836: New changes in PEFT breaks FSDP (RLHF)
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

The FSDP is supported and it works properly with DPO up to `608a2f3` commit tag. However, after this, it crashes.

### Current behaviour

With the latest changes in the main, it takes a very long time before reaching `Map` the dataset. Then it crashes.

### Steps to reproduce

- launch a RunPod instance with 8x A100 or H100 with 640G memory (or 320G)
- choose `winglian/axolotl-runpod:main-latest` template
- follow these steps
```
rm -rf axolotl
git clone https://github.com/OpenAccess-AI-Collective/axolotl && \
cd axolotl && \
git checkout e299312 && \
pip install setuptools && \
pip install -e .[flash-attn,deepspeed] && \
cd ..
```

- then preprocess and train the config.yaml

### Config yaml

```yaml
base_model: arcee-ai/Arcee-Nova
model_type: AutoModelForCausalLM
tokenizer_type: AutoTokenizer

load_in_8bit: false
load_in_4bit: true
strict: false

save_safetensors: true

rl: dpo
chat_template: chatml
datasets:
  - path: Intel/orca_dpo_pairs
    split: train
    type: chatml.intel

dataset_prepared_path: last_run_prepared
val_set_size: 0.0
output_dir: ./models/Arcee-Nova-DPO-v0.1

adapter: qlora
lora_model_dir:

sequence_len: 1800
sample_packing: false
pad_to_sequence_len: false

lora_r: 64
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:
# lora_modules_to_save:
#   - embed_tokens
#   - lm_head
#lora_target_modules:
#  - gate
#  - q_proj
#  - k_proj
#  - v_proj
#  - o_proj
#  - w1
#  - w2
#  - w3

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

gradient_accumulation_steps: 2
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_torch
lr_scheduler: cosine
learning_rate: 5e-5
train_on_inputs: false
group_by_length: false

bf16: auto
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true
warmup_steps: 50
evals_per_epoch: 1
eval_table_size:
eval_table_max_new_tokens: 128
save_steps: 100
debug:
weight_decay: 0.05
fsdp:
   - full_shard
   - auto_wrap
fsdp_config:
  fsdp_limit_all_gathers: true
  fsdp_sync_module_states: true
  fsdp_offload_params: true
  fsdp_use_orig_params: false
  fsdp_cpu_ram_efficient_loading: true
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
  fsdp_transformer_layer_cls_to_wrap: Qwen2DecoderLayer
  fsdp_state_dict_type: FULL_STATE_DICT
special_tokens:
  pad_token: "<|endoftext|>"
  eos_token: "<|im_end|>"
```

### Possible solution

must checkout prior to the change that broken FSDP: `git checkout 608a2f3`

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

e299312

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

### PR DESCRIPTION
**Pull request by winglian**
Fixes #1836 

@maziyarpanahi

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Thanks for working on this fix. Could the added unit test be used to reproduce the original error (when using axolotl from before the fix)?

**Comment by winglian** [comment]
> Thanks for working on this fix. Could the added unit test be used to reproduce the original error (when using axolotl from before the fix)?

Correct! That's the same configuration I was using to manually test.
